resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -563,6 +565,7 @@ private[yarn] class YarnAllocator(
           }
         }
 
+        allocatorNodeHealthTracker.setSchedulerExcludedNodes(excludeNodes)
```

Comment:
this feels like a weird way to do this based on the YarnAllocatorNodeHealthTracker already reads the excludeNode and has it as a private variable.  This api was meant for excluding nodes that had dynamic failures... ie the scheduler excluded them.  These are not excluded by the scheduler but by user manually.

what if we just have it call refreshExcludedNodes when the YarnAllocatorNodeHealthTracker  is created?  Unless you are trying to dynamically change this config which it wasn't really meant for.